stingray/powerspectrum.py

```diff
@@ -390,3 +389,41 @@ def _make_segment_spectrum(self, lc, segment_size):
             end_ind += nbins
 
         return power_all, nphots_all
+
+
+class DynamicalPowerspectrum(AveragedPowerspectrum):
+    def __init__(self, lc, segment_size, norm="frac"):
+        """
+        Parameters
+        ----------
+        lc : lightcurve.Lightcurve object
+            The time series of which the Dynamical powerspectrum is
+            to be calculated.
+
+        segment_size : float, default 1
+             Length of the segment of light curve, default value is 1 second.
+
+        norm: {"leahy" | "rms"}, optional, default "rms"
+            The normaliation of the periodogram to be used. Options are
+            "leahy" or "rms", default is "rms".
+
+        Attributes
+        ----------
+        matrix : np.ndarray
+            The matrix with fourier frequencies in the first column and time
+            in the second column.
+        """
+        if segment_size < 2*lc.dt:
+            raise ValueError("Length of the segment is too short to form a "
```

Comment:
Oh yes the segment_size should be less than the total length of the light curve and that's it upper limit. But there's also the lower limit of the segment_size which needs to be addressed. Does that sound right? Although I must be adding the check for the upper limit also.